distro_test.go

```diff
@@ -17,75 +16,115 @@ import (
+			d1 := newTestDistro(t, ctx, rootFs)
+			d2 := newTestDistro(t, ctx, rootFs)
+
+			if tc.mockErr {
+				modifyMock(t, func(m *mock.Backend) {
+					m.ShutdownError = true
+				})
+				defer modifyMock(t, (*mock.Backend).ResetErrors)
+			}
+
+			wakeDistroUp(t, d1)
+			wakeDistroUp(t, d2)
+
+			err := wsl.Shutdown(ctx)
+			if tc.wantErr {
+				require.Error(t, err, "Shutdown should have returned an error")
+				return
+			}
+			require.NoError(t, err, "Shutdown should have returned no error")
+
+			requireStatef(t, wsl.Stopped, d1, "All distros should be stopped after Shutdown (d1 wasn't)")
+			requireStatef(t, wsl.Stopped, d2, "All distros should be stopped after Shutdown (d2 wasn't)")
+		})
+	}
 }
 
 func TestTerminate(t *testing.T) {
-	ctx := context.Background()
 	if wsl.MockAvailable() {
-		t.Skip("Skipping test because back-end does not implement it")
-		ctx = wsl.WithMock(ctx, mock.New())
+		t.Parallel()
+	}
+
+	testCases := map[string]struct {
+		mockErr bool
+
+		wantErr bool
+	}{
+		"Success": {},
+
+		// Mock-induced errors
+		"Error because wsl.exe returns an error": {mockErr: true, wantErr: true},
 	}
 
-	sampleDistro := newTestDistro(t, ctx, rootFs)  // Will terminate
-	controlDistro := newTestDistro(t, ctx, rootFs) // Will not terminate, used to assert other distros are unaffected
+	for name, tc := range testCases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			ctx, modifyMock := setupBackend(t, context.Background())
+			testDistro := newTestDistro(t, ctx, rootFs)
+			controlDistro := newTestDistro(t, ctx, rootFs)
 
-	defer startTestLinuxProcess(t, &sampleDistro)()
-	defer startTestLinuxProcess(t, &controlDistro)()
+			if tc.mockErr {
+				modifyMock(t, func(m *mock.Backend) {
+					m.TerminateError = true
+				})
+				defer modifyMock(t, (*mock.Backend).ResetErrors)
+			}
+
+			wakeDistroUp(t, testDistro)
+			wakeDistroUp(t, controlDistro)
 
-	err := sampleDistro.Terminate()
-	require.NoError(t, err, "Unexpected error attempting to terminate")
+			err := testDistro.Terminate()
+			if tc.wantErr {
+				require.Error(t, err, "Terminate should have returned an error")
+				return
+			}
+			require.NoError(t, err, "Terminate should have returned no error")
 
-	require.False(t, isTestLinuxProcessAlive(&sampleDistro), "Process was not killed by termination.")
-	require.True(t, isTestLinuxProcessAlive(&controlDistro), "Process was killed by termination of a different distro.")
+			requireStatef(t, wsl.Stopped, testDistro, "The test distro should be stopped after terminating it")
+			requireStatef(t, wsl.Running, controlDistro, "The control distro should be running after terminating the test distro")
+		})
+	}
 }
 
-// startTestLinuxProcess starts a linux process that is easy to grep for.
-func startTestLinuxProcess(t *testing.T, d *wsl.Distro) context.CancelFunc {
+// wakeDistroUp is a test helper that launches a short command and ensures that the distro is running.
+func wakeDistroUp(t *testing.T, d wsl.Distro) {
```

Comment:
I generally think the helper functions should be after all tests, so that we focus on test, and can jump if we need those